code/tutorial_code/tutorials_12/base/contains.sl

```diff
@@ -0,0 +1,14 @@
+namespace: tutorials_12.base
+
+decision:
+  name: contains
+
+  inputs:
+    - container:
+        default: ""
+        required: false
+    - sub
+
+  results:
+    - DOES_NOT_CONTAIN: ${container.find(sub) == -1}
```

Comment:
minor: how about putting CONTAINS as first result with something like this `${container.find(sub) >= 0}` - that way you are not coupled to the exact negative value the function returns